metrics/storage_collector.go

```diff
@@ -0,0 +1,85 @@
+package metrics
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog"
+)
+
+type StorageSizeCollector struct {
+	storageDir  string
+	storageSize prometheus.Gauge
+	//TODO: think of adding error metric indicating we couldn't update the db size
+	interval time.Duration
+	logger   zerolog.Logger
+}
+
+func NewStorageSizeCollector(logger zerolog.Logger, storageDir string, interval time.Duration) (*StorageSizeCollector, error) {
+	storageSize := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "storage_size_bytes",
+			Help: "Estimated disk usage of storage in bytes",
+		})
+
+	if err := prometheus.Register(storageSize); err != nil {
+		logger.Err(err).Msg("failed to register metric")
+		return nil, err
+	}
+
+	return &StorageSizeCollector{
+		storageDir:  storageDir,
+		storageSize: storageSize,
+		interval:    interval,
+		logger:      logger,
+	}, nil
+}
+
+func (c *StorageSizeCollector) Start(ctx context.Context) {
+	go func() {
```

Comment:
good idea to add a defer to recover from any potential panics, since this is not a critical component it shouldn't crash the node in case it doesn't work